ios/Classes/TiMapView.m

```diff
@@ -895,21 +895,47 @@ - (TiMapAnnotationProxy*)proxyForAnnotation:(MKAnnotationView*)pinview
 	return nil;
 }
 
-- (void)mapView:(MKMapView *)mapView didSelectAnnotationView:(MKAnnotationView *)view{
-	if ([view conformsToProtocol:@protocol(TiMapAnnotation)])
-	{
+- (void)mapView:(MKMapView *)mapView didSelectAnnotationView:(MKAnnotationView *)view
+{
+    if ([view conformsToProtocol:@protocol(TiMapAnnotation)])
+    {
+        BOOL isSelected = [view isSelected];
+        MKAnnotationView<TiMapAnnotation> *ann = (MKAnnotationView<TiMapAnnotation> *)view;
         
-		BOOL isSelected = [view isSelected];
-		MKAnnotationView<TiMapAnnotation> *ann = (MKAnnotationView<TiMapAnnotation> *)view;
-		[self fireClickEvent:view source:isSelected?@"pin":[ann lastHitName]];
+        selectedAnnotation = ann;
+        
+        //if canShowCallout == true we will try to find calloutView to hadleTap on callout
+        if ([ann canShowCallout]) {
+            //run after delay because we have to wait for animation finish
+            [self performSelector:@selector(findCalloutView:) withObject:ann afterDelay:0.2];
```

Comment:
From Apple SDK we have an event "click on pin" but don't have an event "callout showed and all subviews in it are already created". 

So we wait 200ms of animation after "click on pin" event and then try to find subview which class is "_MKSmallCalloutPassthroughButton" recursively in callout views hierarchy. 

It is not a "SDK way" - it is a "hack".

Do you have any idea how to do it more "nice" without fixed timeouts? I don't know :( and didn't find any ideas in the net too.